### ISSUE TITLE
Consider adding a method to batch normalise a collection of Jacobian points

### ISSUE THREAD

**Issue by kevaundray** [ConsenSys/gnark-crypto#275]
**Problem**

This can be seen as a nice to have. 

Similar to how inversions are expensive, and we use `BatchInvert` to amortise across multiple inversions. Converting from an extended point to an affine point is also relatively expensive because it requires inverting the z coordinate.

**Solution**

One could add a `BatchNormalisation` method to handle this. The code would BatchInvert the Z coordinates of many points and then convert all of those to Affine using the same workflow as the `FromJacobian`.

Example code snippet:

```go

func BatchFromJacobian(jacPoints []curve.G1Jac) []curve.G1Affine {
	numPoints := len(jacPoints)
	normalisedPoints := make([]curve.G1Affine, numPoints)

	zCoordinates := make([]fp.Element, numPoints)
	for i := 0; i < numPoints; i++ {
		zCoordinates[i] = jacPoints[i].Z
	}
	// Note: BatchInvert handles the case when Z is zero
	invertedZCoordinates := fp.BatchInvert(zCoordinates)

	// Rest of code, follows the same logic as FromJacobian
	for i := 0; i < numPoints; i++ {
		// The point at infinity is encoded as (0,0)
		// so the default value of normalised point here is
		// the identity element
		var normalisedPoint curve.G1Affine

		var zInv = invertedZCoordinates[i]

		if zInv.IsZero() {
			normalisedPoints[i] = normalisedPoint
		}

		p1 := jacPoints[i]
		var b fp.Element

		b.Square(&zInv)
		normalisedPoint.X.Mul(&p1.X, &b)
		normalisedPoint.Y.Mul(&p1.Y, &b).Mul(&normalisedPoint.Y, &zInv)

		normalisedPoints[i] = normalisedPoint
	}

	return normalisedPoints
}
```

**Comment by yelhousni** [comment]
@kevaundray I think it's already implemented in `BatchJacobianToAffineG1()` for all curves, as it is used in `BatchScalarMultiplicationG1()`. But only for G1, Do you need it for G2 as well?
